articles/changes.md

```diff
@@ -0,0 +1,257 @@
+In ROS 2 a Python package can use anything in setup.py files, e.g. entry points since they are being invoked with `python3 setup.py install`.
+
+
+### Environment setup
+
+In ROS 1 the build tool generates scripts which must be sourced in order to setup the environment before being able to use the built ROS packages.
+This approach only works when the ROS packages are being built with ROS specific build tool.
+
+In ROS 2 the environment setup is separated into package-specific scripts and workspace-specific scripts.
+Each package provides the necessary scripts to make itself usable after being built.
+The build tool only invokes the workspace-specific scripts which then call the package-specific scripts.
+
+
+### No non-isolated build
+
+In ROS 1 multiple packages can be built in a single CMake context.
+While this speeds up the build step, every package needs to ensure that cross package target dependencies are defined correctly.
+Additionally all packages share the same namespace which leads to colliding target names, etc.
+
+In ROS 2 only isolated builds are supported, i.e. every package is built separately on its own.
+
+
+### No devel space
+
+In ROS 1 packages can be built without installing them.
+From the devel space in combination with the source space the system is already usable.
+But every package has to actively support the devel space, e.g. in environment hooks and CMake code.
+
+In ROS 2 a package must be installed after building it before it can be used.
+
+One reason for the devel space in ROS 1 is to enable the developer to change files, e.g. Python code or launch files, and use the modified code directly without the need to rebuild the package.
+This benefit is preserved in ROS 2 by optionally replacing copy operations in the install steps with symlinks.
+
+
+### Support catkin_simple use case
+
+In ROS 1 the package catkin_simple is aiming to make writing the CMake code of ROS packages easier.
+In many cases it is not achieving this goal which is often due to restrictions of the design necessary for support features like the devel space.
+
+In ROS 2 the CMake API was restructured to support this use case.
+
+
+### Minimal support for packages without manifest
+
+In ROS 1 only packages with a manifest file are considered by the build system.
+In ROS 2 it is possible to detect packages with supported build system in folders without a manifest file.
+If the package follows common practice it might even be possible to detect some of the missing meta information (like dependencies).
+
+
+## Messages, Services
+
+For more information please see the [ROS interface definition](http://design.ros2.org/articles/interface_definition.html) article.
+
+
+### Separated namespaces
+
+In ROS 1 .msg and .srv files can have the same name but the generated code collides.
+The same is the case for the request and response parts of services.
+
+In ROS 2 the generated code uses separate namespaces to guarantee it is collision-free.
```

Comment:
We have heard your concern. The current ROS 1 <-> ROS 2 bridge is a pretty practical example though where this is necessary.

This article only describes what the current changes are. We will for sure revisit this when looking closer into migration and compatibility layers.